blocks/extensions/training.py

```diff
@@ -0,0 +1,28 @@
+from blocks.extensions import TrainingExtension
+
+
+class AdjustParameter(TrainingExtension):
+    """Adjusts shared variable parameter using some function.
+
+    Applies function to compute the new value of a shared parameter each
+    iteration.
+
+    This class can be used to adapt over the training process parameters like
+    learning rate, momentum, etc.
+
+    Parameters
+    ----------
+    parameter : shared variable to be adjusted
+    function : function which input number of aggregated examples and outputs
+               the parameter value
+
+    """
+    def __init__(self, parameter, function, **kwargs):
+        super(AdjustParameter, self).__init__(**kwargs)
```

Comment:
You should probably set a default callback here i.e. `kwargs.setdefault("after_every_batch", True)` will make sure this extension is run after each batch by default.